internal/controller/controllers/cluster_controller.go

```diff
@@ -76,14 +105,20 @@ func (r *ClusterReconciler) getPullSecret(ctx context.Context, name, namespace s
 		Name:      name,
 	}
 	if err := r.Get(ctx, key, secret); err != nil {
-		return "", errors.Wrap(err, "to get pull secret")
+		return "", errors.Wrapf(err, "to get pull secret %s", key)
 	}
 
 	data, ok := secret.Data["pullSecret"]
 	if !ok {
 		return "", errors.Errorf("pullSecret is not configured")
 	}
 
+	out, err := base64.StdEncoding.DecodeString(string(data))
+	// if failed to decode return raw data
```

Comment:
For some reason i saw a different behavior  between tests and minikube, in my minikube environment i didn't get it encoded, i tried to define the secret as `stringData` and `data` and in both cases that data wasn't encoded.

Reply:
That matches my understanding. The base64 encoding and decoding happens automatically, so here you should not need to do it manually. I suggest fixing the tests and not having this special case in the code.

Reply:
OK i get it now, test infrastructure is confusing, it doesn't decode the data so i don't need to encode it when i set it. but if set `stringData` it will be a different field, so in tests i need to set data but not encode it.
Anyway fixed.